compiler/rustc_mir_build/src/build/block.rs

```diff
@@ -245,11 +245,6 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
                                                 OutsideGuard,
                                                 true,
                                             );
-                                            this.schedule_drop_for_binding(
-                                                node,
-                                                span,
-                                                OutsideGuard,
-                                            );
```

Comment:
For the beta backport in #102676, I assumed that this call should be removed from where it was before #102257 moved it. That passes your tests, but I would appreciate confirmation that this is correct.

https://github.com/rust-lang/rust/pull/102676/commits/82cca232cf0eb3ca28636e2431ae1aaecb2799c3#diff-e5738be658d33400dd1fd835b714ac43358d070918ca3b80446d45a9e4a426afL236

Reply:
I can confirm this removal is intended. During the investigation, this call introduced duplicated drop glue. The generated MIR from a synchronous function passes validation only because only value drops are emitted in the glue. For async functions, the duplicated drop glue also includes storage drops, which trips the MIR validation failure even after fixing  `build_exit_tree`.

Reply:
To be clear, I was not questioning whether the removal was correct in _this_ pull request. I only want to make sure that it's still correct in a slightly different place in the backport.

Reply:
I see. If you are referring to #102676, specifically [this line](https://github.com/rust-lang/rust/pull/102676/files#diff-e5738be658d33400dd1fd835b714ac43358d070918ca3b80446d45a9e4a426afR235), this is correct. There should be no `schedule_drop_for_binding` after calling `storage_live_binding`.